core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/codeGen/CodeWithConverter.kt

```diff
@@ -7,14 +7,19 @@ import org.jetbrains.kotlinx.jupyter.api.VariableName
  * Class representing generated code declarations for a [Marker].
  *
  * @param declarations The generated code.
+ * @param converter Needs to provide additional info (name) from org.jetbrains.dataframe.impl.codeGen.CodeGenerator to its callers
+ * But at the same time name doesn't make sense for GroupBy where code to be executed contains two declarations
  * @param converter Optional converter for the [Marker], such as a [org.jetbrains.kotlinx.dataframe.api.cast], often used for Jupyter.
  */
-public data class CodeWithConverter(val declarations: Code, val converter: (VariableName) -> Code = EmptyConverter) {
+public data class CodeWithConverter<T : CodeConverter>(
```

Comment:
I see most functions return a `CodeWithWithConverter<*>` and since both `CodeConverterImpl` and `ProvidedCodeConverter` are essentially just a `(VariableName) -> Code`, do we need the generic type `T`?

Reply:
For GroupBy i need to combine names of two generated interfaces together for cast, and so CodeWithConverter<ProvidedCodeConverter> provides this additional info. This parameter doesn't make sense for other usages, so there we use more generic CodeWithWithConverter<*>